src/main/java/com/tiki/server/folder/service/FolderService.java

```diff
@@ -48,6 +57,14 @@ public FolderCreateResponse create(final long memberId, final long teamId,
 		return FolderCreateResponse.from(folder.getId());
 	}
 
+	@Transactional
+	public void delete(final long memberId, final long teamId, final List<Long> folderIds) {
+		memberTeamManagerFinder.findByMemberIdAndTeamIdOrElseThrow(memberId, teamId);
+		List<Folder> folders = folderFinder.findAllById(folderIds, teamId);
+		deleteFolders(folders);
```

Comment:
현재 메서드를 타고타고 들어가서 하나씩 삭제하고 있는데 메서드 하나에 많은 기능이 들어있는 것 같습니다. 
코드가 길어지는 단점이 있을 수는 있지만 그래도 메서드 하나당 하나의 기능만 갖게 하는게 좋지 않을까요?

Reply:
말씀이 이해가 가지 않습니다...

메소드를 타고 타고 들어가서 삭제 -> 메소드 하나가 하나의 기능만 하도록 하기 위함 
(ex. deleteFolders -> 폴더 삭제, deleteChildFolders -> 파라미터로 받은 폴더의 자식 폴더 삭제, deleteDocuments -> 파일 휴지통 이동)

메소드 하나당 하나의 기능만 갖게 함 -> 메소드 길이가 짧아짐.

이라고 생각하는데 민규님 리뷰가 어떤 의민지 이해를 못했습니다...ㅠ

Reply:
deleteFolders(folders)
라는 메서드가 결국 따라오는 하위 폴더 삭제 하위 도큐먼트 삭제도 포함하고 있으니 여러 기능을 포함하고 있는 셈입니다. 아래 코드는 지피티를 돌려서 메서드를 분리한건데 아래와 같은 방식으로 메서드를 분리하는게 어떨까요?  한번 보시고 찬님 생각대로 진행해주시면 좋을것 같습니다!

```suggestion
@Transactional
public void delete(final long memberId, final long teamId, final List<Long> folderIds) {
    memberTeamManagerFinder.findByMemberIdAndTeamIdOrElseThrow(memberId, teamId);
    List<Folder> folders = folderFinder.findAllById(folderIds, teamId);
    folders.forEach(folder -> {
        deleteChildFolders(folder);
        deleteDocuments(folder);
    });
    folderDeleter.deleteAll(folders);
}

private void deleteChildFolders(final Folder folder) {
    List<Folder> childFolders = folderFinder.findAllStartWithPath(folder.getChildPath());
    childFolders.forEach(this::deleteDocuments);
    folderDeleter.deleteAll(childFolders);
}

private void deleteDocuments(final Folder folder) {
    List<Document> documents = documentFinder.findAllByFolderId(folder.getId());
    if (!documents.isEmpty()) {
        deletedDocumentAdapter.save(documents);
        documentDeleter.deleteAll(documents);
    }
}

```

Reply:
저는 한 메소드가 하나의 일만 하는 것이 코드를 깔끔하게 짜는데 큰 기여를 한다고 생각합니다!

그리고 그 과정에서 다른 메소드를 호출하는 것이 코드만 봤을 때는 그 메소드가 하는 일이긴 하지만 위의 하나의 일이라는 개념에는 포함되지 않는다고 생각합니다. 하나의 일만 하기 위해 메소드를 분리하고 이를 호출하는 것은 불가피하다고 생각하기 때문입니다.

개선해주신 코드와 제 코드에서의 차이점은 83번째 줄의 조건문을 제외했을 때 deleteFolders 메소드를 delete로부터 분리했냐 하지 않았냐의 차이라고 생각하는데요.

저는 위 코드는 그러한 점에서 delete라는 메소드에 오히려 많은 로직을 담았다고 생각합니다.

다만 민규님이 제안해주신 코드와 제 코드를 계속해서 비교하는 과정에서 폴더를 삭제하고 그 폴더 속 파일을 삭제하는 로직을 발견해서 이를 수정한 후, 머지하도록 하겠습니다.